adaptived/src/main.c

```diff
@@ -442,6 +442,14 @@ API int adaptived_loop(struct adaptived_ctx * const ctx, bool parse)
 			return ret;
 	}
 
+	pthread_mutex_lock(&ctx->ctx_mutex);
+	rule = ctx->rules;
+	while(rule) {
+		adaptived_dbg("Rule \"%s\" loaded\n", rule->name);
+		rule = rule->next;
+	}
+	pthread_mutex_unlock(&ctx->ctx_mutex);
+
 	pthread_mutex_lock(&ctx->ctx_mutex);
```

Comment:
Can you get rid of lines 451-453? The lock is taken at 445, so no need to grab it again, right?

Reply:
Yes, they could be combined into the same lock.  I wasn't sure how much time it would take to print out the messages, so I was wondering if it would be better to relinquish the lock for a second

But you're right.  That's silly :)